### DISCUSSION TITLE
Display midi output on an OLED display.

### DISCUSSION THREAD

**Discussion by AndersHannigan** [tttapa/Control-Surface#607] [Q&A] [answered]
Hello Pietr/Control Surface community!
Im doing my best to learn this library, but im stuck at one thing. Ive set up a simple controller using 3 potentiometers, and will later on add 9 buttons.

I have an oled display connected through i2c, and I want to display the CC output when a knob is turned/ Button is pressed to an OLED display. How can i do this? The code ive used is down below, and i cant understand how to access the output variables. Thanks for any help!

I´ve tried looking around in the documentation, but as far as i saw, they only showed how to display midi input. Sorry if its a noob question:/

`#include <Control_Surface.h>

USBMIDI_Interface midi;
// Instantiate an array of CCPotentiometer objects
CCPotentiometer potentiometers[] {
  {A0, 0x10},      // {Analog pin connected to potentiometer 1, Controller number of the first potentiometer}
  {A1, 0x11},     
  {A2, 0x12}
};

void setup() {
  Serial.begin(115200);
  Control_Surface.begin(); // Initialize Control Surface
   
}

void loop() {
  Control_Surface.loop(); // Update the Control Surface
  
}`

**Comment by tttapa** [comment, marked as answer]
Most MIDI output elements (including `CCPotentiometer` and `CCButton`) have methods to access the internal state, like `CCPotentiometer::getValue()` and `CCButton::getButtonState()`. You can then print the result of those function calls to the display in whichever way you want.

**Reply by AndersHannigan** [reply]
Thank you very much for the answer! I might be very dumb here, but i cant get it to work. I get the following error when i call the function:

`cannot call member function 'AH::analog_t CS::MIDIFilteredAnalog<Sender>::getValue() const [with Sender = CS::ContinuousCCSender; AH::analog_t = unsigned int]' without object`

Would it be possible to provide a quick example of how you would use the function to Serial.println the CC values? Thank you for all your help!

**Reply by tttapa** [reply]
To print the value of the first potentiometer, you would use:

```cpp
void loop() {
  Control_Surface.loop();
  Serial.println(potentiometers[0].getValue());
}
```

**Reply by AndersHannigan** [reply]
Thank you so much! Worked like a charm:)
